Cotire Compilation Error

<!-- IF YOU DO NOT FILL THIS TEMPLATE OUT, WE WILL CLOSE YOUR ISSUE! -->

<!-- This template is for problem reports, for feature suggestion etc... feel free to edit it.
 If this is a crash report, upload the crashlog on https://gist.github.com/
 For issues containing a fix, please create a Pull Request following this tutorial: http://www.azerothcore.org/wiki/Contribute#how-to-create-a-pull-request -->


<!-- WRITE A RELEVANT TITLE -->
Cotire Compilation Error


##### SMALL DESCRIPTION:
<!-- Add a one line description of the bug -->
CMake Error at compilation


##### EXPECTED BLIZZLIKE BEHAVIOUR:
<!-- Describe how it should be working without the bug. -->



##### CURRENT BEHAVIOUR:
<!-- Describe the bug in detail. Database to link spells, NPCs, quests etc https://wowgaming.altervista.org/aowow/ -->
azeroth@belisama:~/azerothcore/build$ cmake ../ -DCMAKE_INSTALL_PREFIX=/var/games/wow/server/ -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DTOOLS=0 -DSCRIPTS=1

[ 13%] Building CXX object deps/g3dlite/CMakeFiles/g3dlib.dir/source/UprightFrame.cpp.o
[ 13%] Building CXX object deps/g3dlite/CMakeFiles/g3dlib.dir/source/Vector2.cpp.o
[ 13%] Building CXX object deps/g3dlite/CMakeFiles/g3dlib.dir/source/Vector3.cpp.o
[ 13%] Building CXX object deps/g3dlite/CMakeFiles/g3dlib.dir/source/Vector4.cpp.o
[ 13%] Linking CXX static library libg3dlib.a
[ 13%] Built target g3dlib
[ 13%] Generating CXX prefix source src/common/cotire/common_CXX_prefix.cxx
CMake Warning (dev) at src/common/common_CXX_cotire.cmake:17:
  Syntax Warning in cmake code at column 99

  Argument not separated from preceding token by whitespace.
Call Stack (most recent call first):
  /var/games/wow/azerothcore/deps/cotire/CMake/cotire.cmake:3596 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

[ 13%] Generating CXX prefix header src/common/cotire/common_CXX_prefix.hxx
CMake Warning (dev) at src/common/common_CXX_cotire.cmake:17:
  Syntax Warning in cmake code at column 99

  Argument not separated from preceding token by whitespace.
Call Stack (most recent call first):
  /var/games/wow/azerothcore/deps/cotire/CMake/cotire.cmake:3596 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

[ 13%] Building CXX precompiled header src/common/cotire/common_CXX_prefix.hxx.pch
CMake Warning (dev) at /var/games/wow/azerothcore/build/src/common/common_CXX_cotire.cmake:17:
  Syntax Warning in cmake code at column 99

  Argument not separated from preceding token by whitespace.
Call Stack (most recent call first):
  /var/games/wow/azerothcore/deps/cotire/CMake/cotire.cmake:3596 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

In file included from <built-in>:407:
<command line>:5:19: warning: missing terminating '"' character [-Winvalid-pp-token]
#define _CONF_DIR "
                  ^
<command line>:6:9: error: macro name must be an identifier
#define /var/games/wow/server/etc"" 1
        ^
<command line>:13:15: warning: missing terminating '"' character [-Winvalid-pp-token]
#define NDEBUG" 1
              ^
<command line>:13:15: warning: ISO C99 requires whitespace after the macro name [-Wc99-extensions]
3 warnings and 1 error generated.
CMake Error at /var/games/wow/azerothcore/deps/cotire/CMake/cotire.cmake:1990 (message):
  cotire: error 1 precompiling
  /var/games/wow/azerothcore/build/src/common/cotire/common_CXX_prefix.hxx.
Call Stack (most recent call first):
  /var/games/wow/azerothcore/deps/cotire/CMake/cotire.cmake:3694 (cotire_precompile_prefix_header)


make[2]: *** [src/common/CMakeFiles/common.dir/build.make:64: src/common/cotire/common_CXX_prefix.hxx.pch] Error 1
make[1]: *** [CMakeFiles/Makefile2:890: src/common/CMakeFiles/common.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
--------------------------



##### STEPS TO REPRODUCE THE PROBLEM:
<!-- Describe precisely how to reproduce the bug so we can fix it or confirm its existence:
 - Which commands to use? Which NPC to teleport to?
 - Do we need to have debug flags on Cmake?
 - Do we need to look at the console while the bug happens?
 - Other steps
-->
1. cd azerothcore/build
2. make clean
3. cmake ../ -DCMAKE_INSTALL_PREFIX=/var/games/wow/server/ -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DTOOLS=0 -DSCRIPTS=1
4. make -j 8

##### EXTRA NOTES:
<!--
Any information that can help the developers to identify and fix the issue should be put here.
Examples:
- was this bug always present in AzerothCore? if it was introduced after a change, please mention it
- the code line(s) that cause the issue
- does this feature work in other server appplications (e.g. CMaNGOS, TrinityCore, etc...) ?
-->

##### BRANCH(ES):
<!-- Specify the branch(es) affected by this issue: master, 0.x, 1.x, or another branch. -->

master


##### AC HASH/COMMIT:
<!-- IF YOU DO NOT FILL THIS OUT, WE WILL CLOSE YOUR ISSUE! NEVER WRITE "LATEST", ALWAYS PUT THE ACTUAL VALUE INSTEAD.
Find the commit hash (unique identifier) by running "git log" on your own clone of AzerothCore or by looking at here https://github.com/azerothcore/azerothcore-wotlk/commits/master -->

commit 8061351128463cb41e3a208b733f9a91691adb93



##### OPERATING SYSTEM:
<!-- Windows 7/10, Debian 8/9/10, Ubuntu 16/18 etc... -->

Debian 10

##### MODULES:
<!-- Are you using modules? If yes, list them (note them down in a .txt for opening future issues) -->

mod-ah-bot  mod-eluna-lua-engine  mod-guildhouse  mod-npc-beastmaster  mod-reward-played-time  mod-weekend-xp  mod-who-logged


##### OTHER CUSTOMIZATIONS:
<!-- Are you using any extra script?
 - Did you apply any core patch/diff?
 - Did you modify your database?
 - Or do you have other customizations? If yes please specify them here.
-->

Database minor changes to fit to the modules requirements

<!-- ------------------------- THE END ------------------------------
Thank you for your contribution.
If you use AzerothCore regularly, we really NEED your help to:
 - TEST our fixes ( http://www.azerothcore.org/wiki/Contribute#how-to-test-a-pull-request )
 - Report issues
 - Improve the documentation/wiki
With your help the project can evolve much quicker!
-->


<!-- NOTE: If you intend to contribute more than once, you should really join us on our discord channel! We set cosmetic ranks for our contributors and may give access to special resources/knowledge to them! The link is on our site http://azerothcore.org/
-->


<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/80832054-cotire-compilation-error?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

Still the same issue : 

`<command line>:13:15: warning: ISO C99 requires whitespace after the macro name [-Wc99-extensions]
3 warnings and 1 error generated.
CMake Error at /var/games/wow/azerothcore/deps/cotire/CMake/cotire.cmake:1990 (message):
  cotire: error 1 precompiling
  /var/games/wow/azerothcore/build/src/common/cotire/common_CXX_prefix.hxx.
Call Stack (most recent call first):
  /var/games/wow/azerothcore/deps/cotire/CMake/cotire.cmake:3694 (cotire_precompile_prefix_header)


make[2]: *** [src/common/CMakeFiles/common.dir/build.make:64: src/common/cotire/common_CXX_prefix.hxx.pch] Error 1
make[1]: *** [CMakeFiles/Makefile2:890: src/common/CMakeFiles/common.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
`

--- a/src/cmake/showoptions.cmake
+++ b/src/cmake/showoptions.cmake
@@ -14,7 +14,7 @@ if( UNIX )
 endif()
 
 message("* Install configs to              : ${CONF_DIR}")
-add_definitions(-D_CONF_DIR="\\"${CONF_DIR}\\"")
+add_definitions(-D_CONF_DIR=$<1:"${CONF_DIR}">)
 
 message("")
 

--- a/src/server/worldserver/CMakeLists.txt
+++ b/src/server/worldserver/CMakeLists.txt
@@ -131,7 +131,7 @@ FOREACH(configFile ${MODULE_CONFIG_FILE_LIST})
     get_filename_component(file_name ${configFile} NAME_WE)
     set(CONFIG_LIST ${CONFIG_LIST}${file_name},)
 ENDFOREACH()
-add_definitions(-DCONFIG_FILE_LIST="\\"${CONFIG_LIST}\\"")
+add_definitions(-DCONFIG_FILE_LIST=$<1:"${CONFIG_LIST}">)
 # end handle config file
 
 CU_RUN_HOOK("AFTER_WORLDSERVER_CMAKE")

